File: python-wrapper/src/python/model/embed/ReceiptBlobType.py
from typing import Union, Dict
from enum import Enum

class ReceiptBlobType(Enum):
	Xades = "xades"
	KmehrRequest = "kmehrRequest"
	KmehrResponse = "kmehrResponse"
	SoapRequest = "soapRequest"
	SoapResponse = "soapResponse"
	SoapConversation = "soapConversation"
	Tack = "tack"

	def __serialize__(self) -> object:
		return self.value

	@classmethod
	def _deserialize(cls, data: Union[str, Dict[str, object]]) -> 'ReceiptBlobType':
		if data == "xades":
			return ReceiptBlobType.Xades
		elif data == "kmehrRequest":
			return ReceiptBlobType.KmehrRequest
		elif data == "kmehrResponse":
			return ReceiptBlobType.KmehrResponse
		elif data == "soapRequest":
			return ReceiptBlobType.SoapRequest
		elif data == "soapResponse":
			return ReceiptBlobType.SoapResponse
		elif data == "soapConversation":
			return ReceiptBlobType.SoapConversation
		elif data == "tack":
			return ReceiptBlobType.Tack
		else:
			raise Exception(f"{data} is not a valid value for ReceiptBlobType enum.")

__all__ = ['ReceiptBlobType']

File: cardinal-sdk/src/commonMain/kotlin/com/icure/cardinal/sdk/model/embed/ReceiptBlobType.kt
package com.icure.cardinal.sdk.model.embed

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlin.String

@Serializable
public enum class ReceiptBlobType(
	internal val dtoSerialName: String,
) {
	@SerialName("xades")
	Xades("xades"),

	@SerialName("kmehrRequest")
	KmehrRequest("kmehrRequest"),

	@SerialName("kmehrResponse")
	KmehrResponse("kmehrResponse"),

	@SerialName("soapRequest")
	SoapRequest("soapRequest"),

	@SerialName("soapResponse")
	SoapResponse("soapResponse"),

	@SerialName("soapConversation")
	SoapConversation("soapConversation"),

	@SerialName("tack")
	Tack("tack"),
}
